client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -267,13 +268,18 @@ public boolean insertRecord(
   }
 
   public void triggerPush() throws IOException {
-    asyncPushing = true;
+    try {
+      asyncPushing.acquire();
+    } catch (InterruptedException e) {
+      logger.error("SortBasedPusher thread interrupted while waiting push finished.");
+      TaskInterruptedHelper.throwTaskKillException();
+    }
     dataPusher.checkException();
     executorService.submit(
         () -> {
           try {
             pushData();
-            asyncPushing = false;
+            asyncPushing.release();
```

Comment:
PushData may probably throw runtimeException, then`asyncPushing.release()` better move to finally block.

Reply:
I'm not sure about the original design detail here, but seems it's correct we should abandon the pusher when something is wrong during the push rather than continue.